ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/FSharpResolveUtil.fs

```diff
@@ -81,3 +82,8 @@ let mayShadowPartially (newExpr: ITreeNode) (data: ElementProblemAnalyzerData) (
     |> Seq.collect (fun element -> opens.GetValuesSafe(element.ShortName))
     |> Seq.toArray
     |> OpenScope.inAnyScope newExpr
+
+let isReferenceNotShadowed(checker: FSharpCheckerService, context, name, symbolFullName, opName) =
```

Comment:
What's the purpose of `checker` argument here? Why it's needed to be passed on each call site?

Reply:
Is there any reason to use tupled function form?